testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -162,6 +162,28 @@ def setUp(self):
         self.values['acceptor_resnm'] = np.array([], dtype="<U3")
 
 
+def build_parameters():
+    protein = "protein"
+    nothing = "resname ALA and not backbone"
+    params = []
+    for s1, s2, s1type in itertools.product((protein, nothing),
+                                            (protein, nothing),
+                                            ("donor", "acceptor", "both")):
+        params.append((s1, s2, s1type))
```

Comment:
Must this become a list? Won't `pytest.mark.parametrize` accept the `itertools` generator object?